src/main/java/seedu/address/logic/parser/DeleteCommandParser.java

```diff
@@ -18,12 +19,16 @@ public class DeleteCommandParser implements Parser<DeleteCommand> {
      */
     public DeleteCommand parse(String args) throws ParseException {
         try {
-            Index index = ParserUtil.parseIndex(args);
-            return new DeleteCommand(index);
+            if (args.contains("n/")) {
+                Name name = ParserUtil.parseName(args.replace("n/", "").trim());
+                return new DeleteCommand(name);
+            } else {
+                Index index = ParserUtil.parseIndex(args);
+                return new DeleteCommand(index);
+            }
```

Comment:
Try utilise multiMap to detect n/ instead, remove the try catch no need to catch the parseexception using this too, this also shows exact message of empty name is user just types `delete n/`
```suggestion
ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NAME);
if (arePrefixesPresent(PREFIX_NAME)) {
Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
return new DeleteCommand(name);
} else {
//....
}

private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
    return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
}
                
```